### DISCUSSION TITLE
Mastodon support?

### DISCUSSION THREAD

**Discussion by saschanaz** [LuisMayo/ace-attorney-bot#100] [General]
So I heard you are interested in setting a Mastodon bot. Any update for the plan, as botsin.space will probably not work?

**Comment by LuisMayo** [comment]
Hi.
I'm exploring my options.

My options being mainly two.
1. Making my own Mastodon instance
2. Complying with botsin.space and only tag the caller and the users in the convo who happen to follow the bot.

Both have pros and cons.
- Making my own instance means having to maintain yet another installation on my own. Plus storing the videos myself. Keep in mind that only in Ace Attorney bots I already maintain 3 of them. Plus several non ace attorney bots
- Complying with botsin.space is not ideal since users may have been featured in a video without knowing it.
I'm not sure which one to take to be fair.

On top of that I have yet to tune the RAM usage for the bots. I have some ideas in mind but they'll take some time. My options are vertical-escalating the bot server. Which is not really expensive and I can afford it. OR reworking the bot library into working as a server which would serve the jobs via an internal websockets API. Which is way more complex but it's something  I've been wanting to do for a long time now. Since that way servers CPUs could be more optimally used than how they're currently.

Since I have a full-time job, plus other things in my life, I think it will take me a couple of months before I'm able to launch it.

Again this is open source, I don't have the patent on this. Anyone can take the lead and I'll be happy to aid them. If nobody does it I'll be the one that's for sure

**Comment by LuisMayo** [comment]
P.S: Moving this to a conversation instead of an issue

**Reply by saschanaz** [reply]
Oh I actually opened it as an issue to discuss about how to merge my Mastodon code to this repository, and then forgot about it. I'll file a new issue if that makes sense.

**Reply by LuisMayo** [reply]
Oh sorry then. For the meantime it will be two parallel branches but you're right that we should possibly abstract the API into a component and maintain the same codebase for the both of them
